### PR TITLE
Move pkg/api dependencies on util/subnet to api/util/subnet

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-3574

### What this PR does / why we need it:

Moves `pkg/api/` dependencies of `pkg/util/subnet/` into `pkg/api/util/subnet` so that `pkg/api` does not have cross-package dependencies.

### Test plan for issue:

Unit tests should continue to pass

### Is there any documentation that needs to be updated for this PR?

N/A, refactor

### CODE REVIEW

File: pkg/cluster/deploybaseresources.go [hunk 1]
```diff
@@ -19,10 +19,10 @@ import (
 	utilrand "k8s.io/apimachinery/pkg/util/rand"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/api/util/subnet"
```
Comment (kimorris27) [inline pkg/cluster/deploybaseresources.go:22]: I noticed that in all of the other places that refer to the new `pkg/api/util/subnet`, you renamed the import to `api_subnet`, but not here in `pkg/cluster`. Why is that?

File: pkg/cluster/nsg.go [hunk 2]
```diff
@@ -8,9 +8,9 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/api/util/subnet"
```
Comment (kimorris27) [inline pkg/cluster/nsg.go:11]: See my question in `pkg/cluster/deploybaseresources.go`.

File: pkg/api/v20220904/openshiftcluster_validatestatic_test.go [hunk 3]
```diff
@@ -69,10 +68,9 @@ func validOpenShiftCluster(name, location string) *OpenShiftCluster {
 		Properties: OpenShiftClusterProperties{
 			ProvisioningState: ProvisioningStateSucceeded,
 			ClusterProfile: ClusterProfile{
-				PullSecret:           `{"auths":{"registry.connect.redhat.com":{"auth":""},"registry.redhat.io":{"auth":""}}}`,
-				Domain:               "cluster.location.aroapp.io",
-				Version:              version.DefaultInstallStream.Version.String(),
-				ResourceGroupID:      fmt.Sprintf("/subscriptions/%s/resourceGroups/test-cluster", subscriptionID),
+				PullSecret: `{"auths":{"registry.connect.redhat.com":{"auth":""},"registry.redhat.io":{"auth":""}}}`,
+				Domain:     "cluster.location.aroapp.io",
+				Version:    "4.99.99", ResourceGroupID: fmt.Sprintf("/subscriptions/%s/resourceGroups/test-cluster", subscriptionID),
```
Comment (kimorris27) [inline pkg/api/v20220904/openshiftcluster_validatestatic_test.go:73]: nit: `ResourceGroupID` belongs on its own line.

File: pkg/cluster/deploybaseresources.go (hunk 1 above)
Comment (hawkowl) [reply pkg/cluster/deploybaseresources.go:22]: I think my IDE put that in :D

**Review by ellis-johnson — APPROVED** [review]
LGTM!

**Pull request merged**